test/Datadog.Trace.ClrProfiler.IntegrationTests/AdoNet/MicrosoftDataSqlClientTests.cs

```diff
@@ -59,6 +61,29 @@ public void SubmitsTracesWithNetStandard(string packageVersion)
                 }
             }
         }
+
+        [Fact]
+        [Trait("Category", "EndToEnd")]
+        [Trait("RunOnWindows", "True")]
+        public void SpansDisabledByAdoNetExcludedTypes()
+        {
+            const string dbType = "sql-server";
+            const string expectedOperationName = dbType + ".query";
+
+            SetEnvironmentVariable(ConfigurationKeys.AdoNetExcludeTypes, "System.Data.SqlClient.SqlCommand;Microsoft.Data.SqlClient.SqlCommand;MySql.Data.MySqlClient.MySqlCommand;Npgsql.NpgsqlCommand");
+
+            string packageVersion = PackageVersions.MicrosoftDataSqlClient.First()[0] as string;
+            int agentPort = TcpPortProvider.GetOpenPort();
+
+            using (var agent = new MockTracerAgent(agentPort))
+            using (ProcessResult processResult = RunSampleAndWaitForExit(agent.Port, packageVersion: packageVersion))
+            {
+                Assert.True(processResult.ExitCode >= 0, $"Process exited with code {processResult.ExitCode}");
+
+                var spans = agent.WaitForSpans(1, 2000, operationName: expectedOperationName);
+                Assert.Equal(0, spans.Count);
```

Comment:
NIT: those tests are flawed, because you can't distinguish "I received no span because it worked" versus "I received no span because something went horribly wrong and the tracer completely failed to push traces". If there was a way to send one unrelated span it would greatly improve the test.

Reply:
That's good feedback. I've updated these tests in 1702eee2c95b1fc147b70d6c327ae4b45308b0a9 to collect all spans (which includes our manual spans) and then asserting afterwards. Do you think this is sufficient?